src/math/data1D.cpp

```diff
@@ -197,6 +197,25 @@ const std::vector<double> &Data1D::values() const { return values_; }
 // Return number of values present in whole dataset
 int Data1D::nValues() const { return x_.size(); }
 
+// Return number of non-zero values present in whole dataset
+std::vector<std::pair<double, double>> Data1D::nNonZeroValues() const
+{
+    std::vector<std::pair<double, double>> pairs;
+
+    int idx = 0;
+    std::for_each(std::execution::seq, values_.begin(), values_.end(),
+                  [&pairs, &idx](const auto &val)
+                  {
+                      if (val > 0)
+                      {
+                          pairs.push_back({x_[idx], val});
+                      }
+                      idx++;
+                  });
+
+    return pairs;
+}
+
```

Comment:
```suggestion
// Return number of non-zero values present in whole dataset
std::vector<std::tuple<double, double>> Data1D::nNonZeroValues() const
{
    std::vector<std::tuple<double, double>> pairs;
    auto xys = zip(x_, values_);
    
    std::copy_if(std::execution::par_unseq, xys.begin(), xys.end(), std::back_inserter(pairs), [](const auto &val) { return std::get<1>(val) > 0; });
    return pairs;
}
```

Using zip keeps the x_ and values_ terms cleanly together without needing to keep track of a mutable iterator, which suddenly means that we can do it all in parallel.

Reply:
Sounds ideal